src/bitset.hpp

```diff
@@ -76,63 +93,144 @@ class Bitset {
   // Flip the order of the two sides of a subsplit.
-  Bitset RotateSubsplit() const;
-  size_t SubsplitChunkSize() const;
-  // Get the ith chunk of the subsplit.
-  Bitset SubsplitChunk(size_t i) const;
-  // Get "child 0" (the child bitset with the smaller binary representation)
-  // or "child 1" (the child bitset with the larger binary representation).
-  Bitset SubsplitGetChild(size_t i) const;
-  // Get a string of the bitset in the specified number of chunks, with each chunk
-  // separated by a "|".
-  std::string ToStringChunked(size_t chunk_count) const;
+  Bitset SubsplitRotate() const;
+  // Sorts clades of subsplit so that they are ordered by their taxon representation.
+  Bitset SubsplitSort() const;
+  // Gets the size of each of each clade. This is the same as the size of the whole
+  // taxon set.
+  size_t SubsplitGetCladeSize() const;
+  // Get clade according to its taxon ordering.
+  Bitset SubsplitGetClade(const size_t i) const;
+  // Get clade according to its binary ordering:
+  // Get "clade 0" (the clade bitset with the smaller binary representation)
+  // or "clade 1" (the clade bitset with the larger binary representation).
+  Bitset SubsplitGetOrderedClade(const size_t i) const;
+  // Output subsplit as string of "1" and "0" characters, with each clade separated by a
+  // "|".
   std::string SubsplitToString() const;
-  std::string SubsplitToIndexSetString() const;
-  // Is this the subsplit of a leaf node.
+  // Output subsplit to string as a comma-separated list of true bits positions, with
+  // each clade separated by a "|".
+  std::string SubsplitToVectorOfSetBitsAsString() const;
+  // Is this the subsplit of a leaf node?
   bool SubsplitIsLeaf() const;
-  // Is this the subsplit of a rootsplit.
+  // Is this the subsplit of root node?
+  bool SubsplitIsRoot() const;
+  // Is this the subsplit of a rootsplit?
   bool SubsplitIsRootsplit() const;
-  // Is this the rotated child of the given subsplit?
+  // Is this the rotated clade of the given subsplit?
   bool SubsplitIsRotatedChildOf(const Bitset &other) const;
-  // Is this the sorted child of the given subsplit?
+  // Is this the sorted clade of the given subsplit?
   bool SubsplitIsSortedChildOf(const Bitset &other) const;
-  // Get the union of the two chunks.
-  Bitset SubsplitChunkUnion() const;
-
+  // Get the union of the two clades.
+  Bitset SubsplitCladeUnion() const;
+  // Get whether the given child is the sorted or rotated child to the given parent.
+  static bool SubsplitIsWhichChildOf(const Bitset &parent, const Bitset &child);
+  // Check whether bitset represents set of taxon clades.
+  bool SubsplitIsValid() const;
+  // NNI: Nearest Neighbor Interchange.
+  // Given a parent/child pair of subsplits embedded within a subsplit DAG, produces the
+  // parent/child subsplit pair that results from an NNI operation.
+  static BitsetPair SubsplitNNI(const Bitset &parent_in, const Bitset &child_in,
```

Comment:
In the future, there is the possibility that `NNIOperation` contains more associated data than a node pair.  I was thinking it could hold its contribution to the marginal likelihood (or whatever objective value we use to rank them).